### PR TITLE
now using the separate HTTPS PDB data service

### PR DESCRIPTION
**Pull request by onursumer**
# What? Why?

Now using the separate PDB annotation service instead of portal back-end servlet.

Changes proposed in this pull request:
- Updated PDB annotation service URL to use the separate instance
- Updated MutationMapper to the latest release (v1.1.3)
- Minor update for the mutation id generator (on the server side)
# Checks
- [ ] Runs on Heroku.
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)
- [ ] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to
  hotfix.
# Any screenshots or GIFs?

If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).
# Notify reviewers

Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

### CODE REVIEW

File: portal/src/main/webapp/js/lib/mutationMapper.js [hunk 1]
```diff
@@ -13685,48 +13738,53 @@ function Mutation3dVis(name, options)
...
-
-			if (currentTop > 0)
-			{
-				_prevTop = currentTop;
-			}
-
-			_container.css('top', -9999);
+			_container.hide();
 		}
+
```
Comment (jjgao) [inline portal/src/main/webapp/js/lib/mutationMapper.js]: can we just delete this code?

File: portal/src/main/webapp/js/src/cbio-util.js [hunk 2]
```diff
@@ -468,13 +468,12 @@ cbio.util = (function() {
     function baseMutationMapperOpts() {
         return {
             proxy: {
-                // default pdb proxy are now configured for a separate pdb data source
-                // this is for backward compatibility
+                // configure for the separate pdb-annotation API
```
Comment (jjgao) [inline portal/src/main/webapp/js/src/cbio-util.js]: technically, it's no longer a proxy for the pdb data. Should we rename it?

Comment (onursumer) [inline portal/src/main/webapp/js/src/cbio-util.js]: All data retrieving classes named as Proxy in the MutationMapper, not a good naming convention I guess. Those classes are actually API clients. And this line is required to configure the client to get the data from the new (separate) PDB data service.

File: portal/src/main/webapp/js/lib/mutationMapper.js (hunk 1 above)
Comment (onursumer) [inline portal/src/main/webapp/js/lib/mutationMapper.js]: I think it is okay to delete this. I don't think we will ever go back to Jmol.

**Pull request merged**